pkg/docker/docker.go

```diff
@@ -508,15 +508,11 @@ func runContainerDockerRun(container *docker.Container, d *stiDocker, image stri
 	glog.Infof("\n\n\n\n\nThe image %s has been started in container %s as a result of the --run=true option.  The container's stdout/stderr will be redirected to this command's glog output to help you validate its behavior.  You can also inspect the container with docker commands if you like.  If the container is set up to stay running, you will have to Ctrl-C to exit this command, which should also stop the container %s.  This particular invocation attempts to run with the port mappings %+v \n\n\n\n\n", image, container.ID, container.ID, liveports)
 
 	signalChan := make(chan os.Signal, 1)
-	cleanupDone := make(chan bool)
 	signal.Notify(signalChan, os.Interrupt)
-	go func() {
-		for signal := range signalChan {
-			glog.V(2).Infof("\nReceived signal '%s', stopping services...\n", signal)
-			cleanupDone <- true
-		}
-	}()
-	<-cleanupDone
+
+	// Block until user sends SIGINT.
+	signal := <-signalChan
+	glog.V(2).Infof("\nReceived signal '%s', stopping services...\n", signal)
```

Comment:
Yeah, it has been long enough, I don't remember precisely how the code evolved to this point, though this was certainly one of the areas that received multiple iterations.

Certainly looks cleaner.  And your test does successful.  The command ended and the container stopped.  I do remember that there was some back and forth before about whether the container still showed up with `docker ps -a` ... I do think it went away, so that sounds good too.

Don't have an abundance of experience in this area to say with certainty, but I wonder if the Golang version might have any bearing on how this behaves.  Certainly we've seen some changes in thread dump processing (which also involves signals) between the versions, as @rhcarvalho noted to us recently.  Perhaps some testing at both 1.4 and 1.5 is warranted?